### DISCUSSION TITLE
Preferred / most accessible way to create a segmented control with react-aria-components?

### DISCUSSION THREAD

**Discussion by schurchleycci** [adobe/react-spectrum#7274] [Q&A] [answered]
It looks like there are a few possible ways to create an [Apple-style segmented control](https://developer.apple.com/design/human-interface-guidelines/segmented-controls) with react-aria-components: using `TagGroup`, `RadioGroup` (if multi-select is not a requirement), `GridList`, `Tabs`, or possibly a `Group` of `ToggleButton`'s. Is there a preferred approach, particularly in terms of accessibility?

**Comment by devongovett** [comment, marked as answer]
We're about to add a ToggleButtonGroup component for this (see #7264), but a RadioGroup also works if you only need single selection. TagGroup and GridList are probably overkill.

Depending on your UI, you could potentially also use Tabs. If the segmented control swaps an associated view between different content, then Tabs would be appropriate. If it's making a selection (like bold/italic/underline buttons), or swapping between views of the same content (e.g. table vs cards of the same items), then ToggleButtonGroup (eventually) or RadioGroup would be more appropriate.

**Reply by schurchleycci** [reply]
It'll be swapping between views of the same content, so `RadioGroup` it is until `ToggleButtonGroup` is released. Thanks @devongovett !
